tests/integration/components/aggregate-query-maker-test.js

```diff
@@ -19,36 +19,46 @@ test('it renders', function(assert) {
 
   this.render(hbs`{{aggregate-query-maker startDate="2010-06-01" endDate="2016-07-02" submit=(action mockSubmit) reset=(action mockReset)}}`);
 
-  assert.notEqual(this.$().text(), '');
+  assert.notEqual(this.$().text(), '', 'The component rendered correctly.');
 
 });
 
-/*
-
-test('it submits the query params', function(assert) {
-  const universityVillage = {"type":"Feature","properties":{},"geometry":{"type":"Polygon","coordinates":[[[-87.67248630523682,41.86454328565965],[-87.67248630523682,41.872117384500754],[-87.6549768447876,41.872117384500754],[-87.6549768447876,41.86454328565965],[-87.67248630523682,41.86454328565965]]]}};
-  let mockLayer = L.geoJson(universityVillage);
-
-  this.set('mockSubmit', (actual) => {
-    let expected = {
-      geom: JSON.stringify(mockLayer.toGeoJSON()),
-      startDate: '2016-2-10',
-      endDate: '2016-2-10',
-      agg: 'day'
-    };
-    assert.equal(actual.geoJSON, expected.geom, 'Failed to report correct geoJSON');
-    assert.equal(actual.agg, expected.agg, 'Failed to report selected aggregation unit.');
+test('Parameters correctly control the component.', function(assert){
+  this.set('start', '2010-06-10');
+  this.set('end', '2016-07-20');
+  this.set('agg', 'day');
+  this.set('center', 'chicago');
+
+  this.set('mockSubmit', () => {
+    // NoOp
   });
 
   this.set('mockReset', () => {
     // NoOp
   });
 
-  this.set('layer', mockLayer);
+  this.set('cities', {
+    "chicago": {label: "Chicago", location: [41.795509, -87.581916], zoom: 10},
+    "newyork": {label: "New York", location:[40.7268362,-74.0017699], zoom: 10},
+    "seattle": {label: "Seattle", location:[47.6076397,-122.3258644], zoom: 11},
+    "sanfrancisco": {label: "San Francisco", location:[37.7618864,-122.4406926], zoom: 12},
+    "austin": {label: "Austin", location:[30.3075693,-97.7399898], zoom: 10},
+    "denver": {label: "Denver", location:[39.7534338,-104.890141], zoom: 11},
+    "bristol": {label: "Bristol, UK", location:[51.4590572,-2.5909956], zoom: 11}
+  });
 
-  this.render(hbs`{{aggregate-query-maker geoJSON=layer submit=(action mockSubmit) reset=(action mockReset)}}`);
-  this.$('#agg-select').val('day').change();
+  this.set('aggOptions', [
+    {id: 'day', label: 'day'},
+    {id: 'week', label: 'week'},
+    {id: 'month', label: 'month'},
+    {id: 'quarter', label: 'quarter'},
+    {id: 'year', label: 'year'}
+  ]);
 
-  this.$('#submit-query').click();
-});
-*/
+  this.render(hbs`{{aggregate-query-maker startDate=start endDate=end agg=agg center=center cities=cities aggOptions=aggOptions centerCoords=centerCoords submit=(action mockSubmit) reset=(action mockReset)}}`);
+
+  assert.equal($("#start-date-filter .form-control").val(), "06/10/2010", "Correctly rendered component based on initial startDate parameter.");
```

Comment:
Other than the above complaint, this test is very clear. Set params, assert those params got rendered. Very nice.